backend/src/main/java/tqs/medex/service/ProductService.java

```diff
@@ -36,4 +36,27 @@ public Product addNewProduct(ProductPOJO productPOJO) {
     product.setSupplier(supplier);
     return productrepository.save(product);
   }
+
+  public Product updateProduct(long productid, ProductPOJO productPOJO) {
+    var product = productrepository.findById(productid).orElse(null);
+    if (product == null) {
+      return null;
```

Comment:
Would it be better to return null or throw an Error? Like ProductNotFoundException for example.

Reply:
Throwing Exceptions as errors in services is not really a good idea, since the controllers where these services are called, need to implement a "Try, Catch" Mechanism, which is a little bit slower.. Better to throw exceptions in controllers than in services.

Reply:
Okay, I'll change the implementation of the services I wrote to match yours.